broker/api.py

```diff
@@ -293,8 +302,22 @@ def update(
                 validators.ErrorResponseConfig(instance.error_responses).validate()
 
             queue = queue_all_cdn_update_tasks_for_operation
-        else:
+        elif instance.instance_type == "alb_service_instance":
+            if details.plan_id != ALB_PLAN_ID:
+                raise ClientError("Updating service plan is not supported")
             queue = queue_all_alb_update_tasks_for_operation
+        elif instance.instance_type == "migration_service_instance":
+            if details.plan_id == CDN_PLAN_ID:
+                noop = False
+                validate_migration_to_cdn_params(params)
+                instance = change_instance_type(
+                    instance, CDNServiceInstance, db.session
+                )
+                update_cdn_params_for_migration(instance, params)
+                db.session.add(instance.current_certificate)
```

Comment:
Is the current certificate being added explicitly here only in the case of migrations because it's needed during that time?

Furthermore, `update_cdn_params_for_migrations` sets a bunch of other things based on the `params` passed in if this code path is reached, but does anything actually happen with those values?